order/utils.py

```diff
@@ -14,30 +14,36 @@ def apply_discount(discount, cost):
 def calculate_discounted_cost(obj, *args, **kwargs):
     # the function applies all discounts and selects the one 
     # with the highest benefit until all discounts are over
-    if obj.weight and obj.discount:
+    if obj.weight and obj.discounts:
         excluded_ids = []
-        discounts = obj.discount.all()
+        discounts = obj.discounts.all()
         our_cost = float(obj.cost)
+        loop = True
 
-        while discounts:
+        while discounts and loop:
```

Comment:
Not nice way of coding. Consider refactoring by splitting code chunks into helper functions